api/v1beta1/moduleimagesconfig_types.go

```diff
@@ -42,6 +42,9 @@ type ModuleImageSpec struct {
 	// kernel version for which this image is targeted
 	KernelVersion string `json:"kernelVersion"`
 
+	// selector for pod that should run builds
+	Selector map[string]string `json:"selector,omitempty"`
```

Comment:
We already have a selector in the build object. Why do we need another one?

Reply:
the selector should be taken from mld: its value is decided based on Selector in Spec and Selector in Build. If Selector in Build is present, then it is taken, otherwise the Spec's Selector is used

Reply:
Then I think its the `Module` controller's job to update the MIC.build.selector (to be passed to MBSC) with the correct value (from module's spec/selector).

I don't think we should add a new selector in MIC and MBSC just to take that decision in MBSC controller. Let's just create the correct build object in MIC to begging with.

WDYT?